### PR TITLE
Add Google Map to About Me tab

### PR DESCRIPTION
**Pull request by tttzach**
Basic functioning Google Map centered at the University of Waterloo added to the About Me tab.

Screenshot 2020-06-08 at 5 29 47 PM

### CODE REVIEW

**Comment by tttzach** [comment]
Change default map type to Terrain

Screenshot 2020-06-08 at 6 38 04 PM

**Comment by tttzach** [comment]
Add info window which appears when marker is clicked

Screenshot 2020-06-08 at 6 38 23 PM

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -92,3 +92,58 @@ function createCommentElement(comment) {
...
+        '<p>The <b>University of Waterloo</b> (commonly referred to as <b>Waterloo</b>, <b>UW</b>, or '+
+        '<b>UWaterloo</b>) is a public research university with a main campus in Waterloo, Ontario, '+
+        'Canada. The main campus is on 404 hectares of land adjacent to Uptown Waterloo and Waterloo '+
+        'Park. The university also operates three satellite campuses and four affiliated university '+
+        'colleges.</p>'+
+        '<p>Attribution: University of Waterloo, <a href="https://en.wikipedia.org/wiki/University_of_Waterloo">'+
+        'https://en.wikipedia.org/wiki/University_of_Waterloo</a> '+
+        '(last visited June 8, 2020).</p>'+
+        '</div>'+
+        '</div>';
```
Comment (gbuenoandrade) [inline portfolio/src/main/webapp/script.js:142]: Nit: You can have multiline strings using [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) (they do various cool things apparently https://css-tricks.com/html-templates-via-javascript-template-literals/)

File: portfolio/src/main/webapp/libraries.js [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  });
+  marker.addListener('click', function () {
+    infoWindow.open(map, marker);
+  });
+}
+
+google.charts.load('current', { 'packages': ['corechart'] });
+google.charts.setOnLoadCallback(drawPieChart);
+google.charts.setOnLoadCallback(drawRegionsChart);
+google.charts.setOnLoadCallback(drawCoronavirusChart);
```
Comment (gbuenoandrade) [inline portfolio/src/main/webapp/libraries.js:72]: Nit: Maybe move these to the top of the file?

File: portfolio/src/main/webapp/libraries.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    'width': 700,
+    'height': 500
+  };
+
+  const chart = new google.visualization.GeoChart(document.getElementById('regions-chart'));
+  chart.draw(data, options);
+}
+
+function drawCoronavirusChart() {
+  fetch('/coronavirus-data').then(response => response.json())
```
Comment (gbuenoandrade) [inline portfolio/src/main/webapp/libraries.js:116]: Nit: Break the line before this `.then` to be consistent with the `then` at line 117

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (gbuenoandrade) [inline portfolio/src/main/webapp/script.js:142]: I think you no longer need this file?

File: portfolio/src/main/webapp/libraries.js (hunk 2 above)
Comment (tttzach) [reply portfolio/src/main/webapp/libraries.js:72]: Good call, thanks!

File: portfolio/src/main/webapp/libraries.js (hunk 3 above)
Comment (tttzach) [reply portfolio/src/main/webapp/libraries.js:116]: Thanks for catching that, I'll be more careful.

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (tttzach) [reply portfolio/src/main/webapp/script.js]: Yes you're right, I will be more careful next time!

**Comment by tttzach** [comment]
I have implemented the suggestions, though I am not sure why some changes from Google Charts are also in this PR. I will keep an eye out to see if it happens again or it might be a mistake on my part. Thanks for the reviews Gui!

File: portfolio/src/main/webapp/libraries.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  const marker = createMapMarker(map, uWaterloo);
+  changeMapToTerrain(map);
+  createMapInfoWindow(map, marker);
+}
+
+function mapInit(position) {
+  const map = new google.maps.Map(
+    document.getElementById('map'), {
+    center: position,
+    zoom: 16
```
[the remark this answers is not in the document]
Comment (tttzach) [reply portfolio/src/main/webapp/libraries.js]: Thanks for the suggestion, I've implemented it here #42.

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -41,7 +44,12 @@ <h1>definitely not zachary's website.</h1>
   <div id="about-me" class="containerTab">
     <span onclick="this.parentElement.style.display='none'" class="closebtn">x</span>
     <h2>About Me</h2>
-    <p>Hello! My name is Zachary, I am a rising junior at the University of Waterloo, Canada pursuing a major in Combinatorics & Optimization and a minor in Computer Science.<br/>Feel free to click on the other tabs for more not-so-exciting information about me.
+    <p>Hello! My name is Zachary, I am a rising junior at the University of Waterloo, Canada pursuing a major
+      in Combinatorics & Optimization and a minor in Computer Science.<br/>Feel free to click on the other
+      tabs for more not-so-exciting information about me.
+    <br/><br/>
```
[the remark this answers is not in the document]
Comment (tttzach) [reply portfolio/src/main/webapp/index.html]: You're absolutely right, I've changed it here #43. I know there's more `<br/><br/>` around the comments form but I would like to do the reformatting on a different PR so I'll get used to the workflow you suggested which I think is incredibly helpful. Thanks!

File: portfolio/src/main/webapp/libraries.js [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+google.charts.setOnLoadCallback(drawRegionsChart);
+google.charts.setOnLoadCallback(drawCoronavirusChart);
+
+// Coordinates of the University of Waterloo from manually checking Google Maps
+const uWaterloo = { lat: 43.473, lng: -80.545 };
+// Show all the main buildings of the University of Waterloo
+const zoom = 16;
+// Match height and width to other visual elements on the About Me tab
+const defaultWidth = 700;
+const defaultHeight = 500;
```
Comment (gbuenoandrade) [inline portfolio/src/main/webapp/libraries.js:26]: Nit: Use CONSTANT_CASE for those (https://google.github.io/styleguide/jsguide.html#naming-constant-names)

File: portfolio/src/main/webapp/libraries.js [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+  createMapInfoWindow(map, marker);
+}
+
+function mapInit(position) {
+  const map = new google.maps.Map(
+    document.getElementById('map'), {
+    center: position,
+    zoom: zoom
+  }
+  );
```
Comment (gbuenoandrade) [inline portfolio/src/main/webapp/libraries.js:41]: Nit: Move those symbols to line 40 `});` (https://google.github.io/styleguide/jsguide.html#formatting-function-expressions)

Comment (tttzach) [reply portfolio/src/main/webapp/libraries.js]: Thanks! I've changed it in the last commit.

File: portfolio/src/main/webapp/libraries.js (hunk 6 above)
Comment (tttzach) [reply portfolio/src/main/webapp/libraries.js]: Thanks for catching this, my bad!

**Pull request merged**